src/canopy_dealloc.F90

```diff
@@ -59,6 +59,18 @@ SUBROUTINE canopy_dealloc
     if(allocated(ppfd_shade240_tmp))     deallocate(ppfd_shade240_tmp)
     if(allocated(tmp2mref_tmp))          deallocate(tmp2mref_tmp)
     if(allocated(ubzref_tmp))            deallocate(ubzref_tmp)
+    if(allocated(tleaf_sun24_rst))       deallocate(tleaf_sun24_rst)
```

Comment:
Similar comment here and below for having new "rst" variables. Are these really needed>?